krabs/krabs/ut.hpp

```diff
@@ -19,7 +19,13 @@ namespace krabs { namespace details {
     struct ut {
 
         typedef krabs::provider<> provider_type;
+        
+        struct filter_settings{
+            std::vector<unsigned short> provider_filter_event_ids_;
+            std::tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR> flags_tuple_;
+        };
 
+        typedef std::map<krabs::guid, filter_settings> provider_filter_settings;
```

Comment:
Should this be `map` or `unordered_map`? `unordered_map` is ultimately a Hash Map while `map` will be a BTree Map.

It doesn't look like order matters so I'd opt for `unordered_map`.

Reply:
Done

Reply:
Hi, Zac

This modification will cause additional changes and I am not sure it is the time doing them. Originally was implemented as map before my modifications. I just moved it being typedef.

I guess why originally implemented as map because krabs::guid is not supporting option for hashing for being a key of unordered collections. Such change will require me adding hashing for guid which will involve GUID conversion to string and hasing afterwards. In this case, seems to me it is very small map so effective performance will not be influenced by converting it to unordered_map
Please let me know if you insist on unordered map implementation.
it will cause changes in guid.hpp file:
adding:
 
namespace std
{
	template <>
	struct hash<krabs::guid>
	{
        typedef krabs::guid argument_type;
        typedef std::size_t result_type;

        result_type operator()(const argument_type &guid) const {
            return hash<std::wstring>()(to_wstring(guid));
        }
    };
}

Thanks
Issahar

Reply:
Not fixed: Please reopen the issue upon response